lib/burnside/flattened.cc

```diff
@@ -705,17 +705,10 @@ struct ControlFlowAnalyzer {
 
 }  // namespace flat
 
-template<typename A>
-void CreateFlatIR(const A &ptree, std::list<flat::Op> &ops, AnalysisData &ad) {
+void CreateFlatIR(std::list<flat::Op> &ops, AnalysisData &ad) {
   flat::ControlFlowAnalyzer linearize{ops, ad};
-  Walk(ptree, linearize);
+  std::visit(
+      [&](const auto *ptree) { Walk(*ptree, linearize); }, ad.parseTreeRoot);
```

Comment:
This seems to be downshifting the root node types out of the template arguments (type space) to a variant data structure, but I wonder if there isn't a way to call the right return processing without that step? On the other hand, there is time to compile ...

Reply:
You are correct, this technically adds a dynamic overhead to switch through the variant, but as it is a pretty high level function and the variant is small, I thought it was worth the simplicity.